crates/node/src/commands/setup.rs

```diff
@@ -0,0 +1,176 @@
+use std::path::{Path, PathBuf};
+
+use sc_cli::{CliConfiguration, Error, Result, SharedParams, SubstrateCli};
+use sc_service::BasePath;
+use url::Url;
+
+use crate::chain_spec::GENESIS_ASSETS_DIR;
+use crate::cli::Cli;
+use crate::configs::FileInfos;
+use crate::{configs, constants};
+
+/// Define a way to retrieve an index.json file
+
+/// The index.json must follow the format of the official index.json
+/// (https://github.com/keep-starknet-strange/madara/blob/main/configs/index.json)
+/// Where the `md5` and `url` fields are optional
+#[derive(Debug, clap::Args)]
+#[group(required = true, multiple = false)]
+pub struct SetupSource {
+    /// Download an index.json file for an url
+    #[clap(
+        long,
+        conflicts_with="from_local",
+        value_hint=clap::ValueHint::Url,
+        // This combination of properties allow us to use a default value only if the arg is passed without value.
+        // If it is not passed at all, no default value is used.
+        // See: https://docs.rs/clap/latest/clap/struct.Arg.html#method.default_missing_value
+        num_args = 0..=1, require_equals=true, default_missing_value = Some(constants::DEFAULT_CONFIGS_URL)
+    )]
+    pub from_remote: Option<String>,
+
+    /// Copy an index.json file for an url
+    #[clap(long, conflicts_with = "from_remote")]
+    pub from_local: Option<String>,
+}
+
+#[derive(Debug, clap::Args)]
+pub struct SetupCmd {
+    #[allow(missing_docs)]
+    #[clap(flatten)]
+    pub shared_params: SharedParams,
+
+    #[clap(flatten)]
+    pub source: SetupSource,
+}
+
+impl CliConfiguration for SetupCmd {
+    fn shared_params(&self) -> &SharedParams {
+        &self.shared_params
+    }
+}
+
+impl SetupCmd {
+    pub fn run(&self) -> Result<()> {
+        log::info!("setup cmd: {:?}", self);
+        let dest_config_dir_path = {
+            let is_dev = self.shared_params().is_dev();
```

Comment:
shouldn't it be verified by the `--chain dev`?

Reply:
You are right. Because we overwrite the `is_dev` setting.

Reply:
well actually no. We overwrite it only for run.
Not for setup

Reply:
yes, the argument is that `--dev` shouldn't exist in this context at all, and we should just used the `--chain` flag

Reply:
`--chain dev` and `--chain local --dev` are two different things

the first one give you the `dev` chain
the second one the `local` with `force_authorign` and `alice` and rpc_exteranal unsafe

Reply:
it should be
```
    if cmd.base.shared_params.chain.is_none() {
        cmd.base.shared_params.chain = Some("dev".to_string())
    };
```
in the override method

Reply:
oh no you are right

```rust
	#[arg(long, conflicts_with_all = &["chain"])]
	pub dev: bool,
```

Reply:
but still, you can run the following `--chain=dev` without `--alice` and so on

And the impl of `.chain_id(is_dev)` is the following:

```rust
	/// Get the chain spec for the parameters provided
	pub fn chain_id(&self, is_dev: bool) -> String {
		match self.chain {
			Some(ref chain) => chain.clone(),
			None =>
				if is_dev {
					"dev".into()
				} else {
					"".into()
				},
		}
	}
```

Reply:
so no. It should stay as it is

Reply:
> yes, the argument is that --dev shouldn't exist in this context at all, and we should just used the --chain flag

`--dev` still exists. It's only overwritten for the `run` cmd.

Reply:
You can implement it the way you want, but default behavior for substrate is that `--dev` is and alias for ` --chain dev`

Reply:
oh no, it has a bigger impact.